src/repo/mod.rs

```diff
@@ -1,15 +1,20 @@
 //! IPFS repo
-use crate::block::{Block, Cid};
 use crate::error::Error;
 use crate::path::IpfsPath;
 use crate::IpfsOptions;
 use async_std::path::PathBuf;
 use async_trait::async_trait;
+use bitswap::Block;
 use core::fmt::Debug;
-use futures::channel::mpsc::{channel, Receiver, Sender};
-use futures::SinkExt;
+use core::future::Future;
+use core::marker::PhantomData;
+use core::pin::Pin;
+use core::task::{Context, Poll, Waker};
+use crossbeam::{Receiver, Sender};
```

Comment:
the problem I had and why I changed anything in the first place was because a dag test deadlocked. I'm not familiar enough with the various channel implementations to have a strong opinion, but since crossbeam is already a dependency of async-std I thought I'd use that.

Reply:
Was the deadlock inside futures channels? If so, those stacktraces would be bug reports in `futures` since I am not seeing an open issue related to deadlocks with anything but the way old 0.1.

Reply:
a deadlock doesn't present with a stack trace, it presents with a not terminating test

Reply:
it probably wasn't the futures channel but the fact that it was bounded to 1 and no one was reading from the channel. crossbeam would return an error if there are no listeners anymore? anyway stuff like this is a timesink.

Reply:
> a deadlock doesn't present with a stack trace

Yes, but one can be obtained via attaching gdb for example.

> it was bounded to 1 and no one was reading from the channel. crossbeam would return an error if there are no listeners anymore?

I'd expect the same from futures channels.

> anyway stuff like this is a timesink.

Yes, but at the same time if no-one reported the bugs we encounter the rust ecosystem would be must more brittle.